### ISSUE TITLE
How to do inpainting with vanilla stable diffusion?

### ISSUE THREAD

**Issue by pauldog** [huggingface/diffusers#2017]
Just taking the vanilla 1-4 stable diffusion model, can I get it to do good inpainting?

I can get it to do somewhat good inpainting, by taking the predicted latents and replacing the masked areas with noised latents from the original image at each step.
Also, it helps if I apply a guidance mask to the image so that the original image parts have guidance=1.

However it suffers from sometimes the mask boundary becoming part of the image, 

Is there a better algorithm for inpainting?

**Comment by patrickvonplaten** [comment]
Hey @pauldog,

You can do this by using: https://github.com/huggingface/diffusers/blob/3ecbbd628825712552a1834f78a224bfab4cfe21/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py#L60

The most memory efficient way to do this would be:

```python
from diffusers import DiffusionPipeline, StableDiffusionInpaintPipelineLegacy
import torch

text2img = DiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", torch_dtype=torch.float16)
inpaint = StableDiffusionInpaintPipelineLegacy(**text2img.components)

text2img(...) # to call the function
inpaint(...) # to call the function
```

However note that the "real" inpainting fine-tuned checkpoints are much better: https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/inpaint they do however require you to download two weights.
